src/librepgp/stream-parse.cpp

```diff
@@ -448,7 +485,9 @@ encrypted_start_aead_chunk(pgp_source_encrypted_param_t *param, size_t idx, bool
     size_t  nlen;
 
     /* set chunk index for additional data */
-    STORE64BE(param->aead_ad + param->aead_adlen - 8, idx);
+    if (param->auth_type == rnp::AuthType::AEADv1) {
```

Comment:
Does that mean you also for instance want this code to be duplicated?
``` 
    if (last) {
        uint64_t total = idx * param->chunklen;
        if (idx && param->chunkin) {
            total -= param->chunklen - param->chunkin;
        }

        if (!param->chunkin) {
            /* reset the crypto in case we had empty chunk before the last one */
            pgp_cipher_aead_reset(&param->decrypt);
        }
        STORE64BE(param->aead_ad + param->aead_adlen, total);
        param->aead_adlen += 8;
    }
```

I personally disagree with an approach that leads to code duplication. Code duplication is in my opinion one of the clearest errors in code design, leading to less readability, higher maintenance efforts and more complexity. Also I don't see the benefit of leaving the v5 AEAD code untouched. I think it should suffice if it is clearly visible that in case of v5 AEAD, the code behaves the same.  

If the goal is to have the code a bit cleaner than curently and especially not to create the `add_data_seipd_v2` vector in case of v5 AEAD, then I think we can work towards that without introducing code duplication. I can work on a proposal for that if you agree.

Reply:
@falko-strenzke Nope, duplicating such large code piece would be an overkill. Looks like I clicked on a wrong code line while adding a comment, causing confusion. I mean only that part which builds ad and calls `pgp_cipher_aead_set_ad()`. (so only duplication would be `if (!pgp_cipher_aead_set_ad(&param->decrypt, add_data, add_data_len)) ...`)

Reply:
OK, I came up with a solution which only slightly needs to alter the parts of the original v5 AEAD code. This alteration is necessary in order to be able to handle everything in a single switch block. Coming soon in upcoming commit.